client/gutenberg/editor/edit-post/components/header/header-toolbar/index.js

```diff
@@ -77,7 +77,7 @@ function HeaderToolbar( {
 /* eslint-enable wpcalypso/jsx-classname-namespace */
 
 function getCloseButtonPath( routeHistory, site ) {
-	const editorPathRegex = /^\/gutenberg\/(post|page|(edit\/[^\/]+))\/[^\/]+(\/\d+)?$/i;
+	const editorPathRegex = /^\/gutenberg\/(post|page|edit)($|\/)/i;
```

Comment:
I'm afraid this change breaks the code that few lines later -- the one that uses this regex to extract `editorPostType`. It can be one of `page`, `post` or `edit/type` and the `type` part is then used to construct an "all posts" URL. This new regexp doesn't capture the `type` part.

Reply:
I'm reviewing that in the next days.

Reply:
Thanks @touletan, looking forward to review your contributions! :+1:

Reply:
new proposed solution pushed